pkg/controllers/raycluster_controller.go

```diff
@@ -304,7 +310,10 @@ func isMTLSEnabled(cfg *config.KubeRayConfiguration) bool {
 }
 
 func crbNameFromCluster(cluster *rayv1.RayCluster) string {
-	return cluster.Name + "-" + cluster.Namespace + "-auth" // NOTE: potential naming conflicts ie {name: foo, ns: bar-baz} and {name: foo-bar, ns: baz}
+	if shouldUseHashedName(cluster) {
```

Comment:
I'd suggest to flip the check, so the main / new logic branch is outside the if block.